crates/pallet-domains/src/staking.rs

```diff
@@ -1114,7 +1197,15 @@ pub(crate) fn do_unlock_nominator<T: Config>(
             .ok_or(Error::UnknownNominator)?;
 
         // convert any deposits from the previous epoch to shares
-        do_convert_previous_epoch_deposits::<T>(operator_id, &mut deposit)?;
+        match do_convert_previous_epoch_deposits::<T>(
+            operator_id,
+            &mut deposit,
+            current_domain_epoch_index,
+        ) {
+            // Share price may be missing if there is deposit happen in the same epoch as de-register
+            Ok(()) | Err(Error::MissingOperatorEpochSharePrice) => {}
```

Comment:
Do we need to check that there was actually a deposit and de-register in the same epoch, or is that guaranteed by the code? Is there a test for this happening in both orders? (Deposit then de-register, and de-register then deposit.)

It might also be helpful to document this case in the function the error is generated in.